### DISCUSSION TITLE
building for armel with custom image

### DISCUSSION THREAD

**Discussion by halfbyte** [cross-rs/cross#1296] [Q&A] [answered]
I ran into an issue where I don't know exactly if that's a user error (I'm fairly new to rust and not super experienced with cross compiling) or if this is maybe more of an oversight. 

The container image for the architecture I'm trying to target (`armv5te-unknown-linux-gnueabi`) is based on Ubuntu 20.4, but the issue here is that Ubuntu has no packages available for the `armel` architecture, so I can't follow [the instructions](https://github.com/cross-rs/cross#dockerfiles) to add packages that are needed as build time dependencies (In my particular case: sdl2 libraries).

I wonder what my options are here and if I'm missing something. Wouldn't it be better (if kind of inconvenient for maintainers) to base the docker image on a system (like mainline Debian) that still has packages built for the architecture? 

Does that mean that I need to create my own image (Which I guess would be somewhat doable but would mean I'd need to replicate a lot of code from here)? Maybe such images exist somewhere already?

**Comment by Emilgardis** [comment, marked as answer]
You could definitely try debian, infact shouldn't be hard.

See [this wiki entry](https://github.com/cross-rs/cross/wiki/FAQ#using-debian-repositories)

**Reply by halfbyte** [reply]
Sorry for the late answer, this project has been lying dormant for a while. This did the trick, but the install-deb.sh file required a bunch of changes to work for my case, nothing unsolvable with a bit of debian package management knowledge, though?

(For those interested: Mainly the old debian release used in the script lead to broken packages, I didn't really investigate, but since I wanted to have more up to date packages anyway, I upgraded to `bookworm` and there the security updates and their URLs work a bit differently)
